rust/bin-shared/src/windows.rs

```diff
@@ -36,3 +54,199 @@ pub fn app_local_data_dir() -> Result<PathBuf> {
         .join(crate::BUNDLE_ID);
     Ok(path)
 }
+
+pub fn tcp_socket_factory(addr: &SocketAddr) -> io::Result<TcpSocket> {
```

Comment:
Which binaries is this shared with? Isn't this exclusively used by the headless-client? Unless it is actually shared between binaries, I don't think it should be in `bin-shared`.

The way I see this is that `bin-shared` is primarily useful for firezone-agnostic stuff that is needed by multiple binaries. I don't actually remember why we had to move the `TunDeviceManager` here. Feels like that one should be in headless-client?

Reply:
I think the Gateway uses it too

Reply:
Ah right. That is okay then.